### PR TITLE
#8 change user model

### LINKED ISSUE #8: Change User model and create migrations
Project: #19 
Depends on: #2 

**Description**
To support profiles, additional fields are needed on the User model.  Add the following columns:
* `discord_name` representing their discord `username` + `#` + `discriminator`, e.g. chimten#3819 (string, max 100 chars)
* `bio` representing their bio (string, max 1000 chars, default empty string)
* `twitter_username` (string, max 200 chars, default empty string)
* `github_username` (string, max 200 chars, default empty string)
* `linkedin_url` (string, max 200 chars, default empty string)
* `website_url` (string, max 500 chars, default empty string)

All fields should be non-nullable (required)

When the user is created via Discord (see `server/lib/auth.ts`, we should also be inserting the `discord_name`

**Acceptance Criteria**
* New fields exist on the User model
* Tests pass
* Tests validate that the `discord_name` is generated correctly
* Manually tested

### PR DESCRIPTION
**Pull request by MatthewBozin**
Closes #8 

# Description

Added new fields to user model.
Performed migration for model changes.
Updated auth/user code with Discord username field.
Updated acceptance tests with Discord username field.

## Testing

Describe what a reviewer should do to test your code, including instructions for manually testing as well as relevant automated test suites to run.

Automated: `npm run test:server`
Manual: go through app auth flow

## Type of change

Please select everything applicable.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires an update to testing

# Checklist:

- [x] Changes have new/updated automated tests
- [x] Changes have been manually tested
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: server/lib/auth.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export async function getDiscordUserAndGuilds(authCode: string) {
 export async function upsertUser(user: DiscordOauth2.User) {
   const result = await User.upsert({
     discord_user_id: user.id,
+    discord_name: user.username,
```
Comment (timmyichen) [inline server/lib/auth.ts:54]: So this just saves the username, but doesn't save the discriminator.  When I console.log the user here and log in, I see:
```
{                                              
  id: '437048770237169666',                    
  username: 'chimten',                         
  display_name: null,                          
  avatar: '450a0b622a7c905c4c75e190cb21f2df',  
  avatar_decoration: null,                     
  discriminator: '3819',                       
```

So we want to construct the `discord_name` in the format of `username#discriminator`

**Review by timmyichen — CHANGES_REQUESTED** [review]
Great work! I just have some small comments but otherwise it looks good!

**Comment by MatthewBozin** [comment]
Thanks for the responses @timmyichen, it's always good to follow convent8ion! I've made the requested changes!

**Review by timmyichen — APPROVED** [review]
LGTM!

**Review by wo1vin — APPROVED** [review]
What he said.

File: server/lib/auth.ts [hunk 2]
```diff
@@ -49,6 +49,7 @@ export async function getDiscordUserAndGuilds(authCode: string) {
 }
 
 export async function upsertUser(user: DiscordOauth2.User) {
+  console.log(user);
```
Comment (MainlyColors) [inline server/lib/auth.ts]: Do you want to leave this rogue console.log here?

Comment (MainlyColors) [reply server/lib/auth.ts]: nvm, saw it was removed

**Review by MainlyColors — APPROVED** [review]
LGTM

**Pull request merged**